[docs] [ENG-11843] Update version source docs (#29501)
expo/expo: docs/pages/build-reference/app-versions.mdx

One of the most frequent causes of app store rejections is submitting a build with a duplicate version number. This happens when a developer forgets to increment the version number before running a build.

> EAS Build can manage automatically incrementing these versions for you if you use the "remote" app version source, which is the default behaviour. Optionally, you can choose to use a "local" app version source, which means you control versions manually in their respective config files.

Reviewer: ```suggestion
EAS Build can manage automatically incrementing these versions for you if you use the `remote` app version source, which is the default behaviour. Optionally, you can choose to use a `local` app version source, which means you control versions manually in their respective config files.
```

Remote and local are both properties in **eas.json** so let's use inline syntax highlight here (Unrelated to the changes in the PR).

> > **warning** The "remote" version source as a default behaviour has been introduced in `eas-cli` version `10.0.0`. If you are using an older version of CLI, be aware that **"local" will be the default behaviour**.

Reviewer: ```suggestion
> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `10.0.0`. If you are using an older version of CLI, be aware that **`local` will be the default behavior**.
```

Let's use American spelling as per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#voice-and-tone).

For developer-facing build version, you can set them to autoincrement on every build. This will help you avoid making manual changes to the project every time you upload a new archive on Play Store testing channels or TestFlight. One common cause for app store rejections is submitting a build with a duplicate version number. It happens when a developer forgets to increment the developer-facing build version number before creating a new build.

> EAS Build can help manage developer-facing build versions automatically by incrementing these versions for you if you opt into using the [`remote` version source](#remote-version-source), which is the default behaviour. Optionally, you can choose to use a `local` app version source, which means you control versions manually in their respective config files.

Reviewer 2: I believe that after multiple review rounds we decided that there is no default value (expect for nonInteractive), but the value needs to be specified explicitly

Author: Yes, you're right, the docs change reflect the previous state from before the last reviews

## Remote version source

> > **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, be aware that **`local` will be the default behavior**.

Reviewer: ```suggestion
> **warning** The `remote` version source is the default behavior from EAS CLI version `11.0.0`. If you are using an older version of the EAS CLI, `local` is the default.
```

- Let's revise the callout to be concise to convey the same information.

> EAS servers can store and manage your app's developer-facing build version (`android.versionCode` and `ios.buildNumber`) remotely. This is the default behavior and to enable it you need to set `cli.appVersionSource` to `remote` in **eas.json**. Then, under the `production` build profile, you can set the `autoIncrement` property to `true`.

Reviewer: ```suggestion
EAS servers can store and manage your app's developer-facing build version (`android.versionCode` and `ios.buildNumber`) remotely by default. To enable it, you need to set `cli.appVersionSource` to `remote` in **eas.json**. Then, under the `production` build profile, you can set the `autoIncrement` property to `true`.
```

Let's also mention the default behavior in the first sentence to be concise.

## Local version source

> > **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, you **don't need to specify the version source explicitly to `local`**.

Reviewer: ```suggestion
> **warning** The `remote` version source as a default behavior has been introduced in `eas-cli` version `11.0.0`. If you are using an older version of the CLI, you **don't need to specify the version source explicitly to `local`**.
```

Let's be concise here too based on the previous comment.

> You can also configure your project so that the source of truth for project versions is the local project source code itself. Add `{ "cli": { "appVersionSource": "local" } }` in your **eas.json**. In this case, EAS does not write to the project. It reads the app version values and builds projects as they are. You can opt in to auto incrementing versions locally by setting the `autoIncrement` option on a build profile.

Reviewer: ```suggestion
You can configure your project so that the source of truth for project versions is the local project source code itself. To do this, set `cli.appVersionSource` to `local` in your **eas.json**.

With this setup, EAS reads app version values and builds projects as they are. It doesn't write to the project. You can also enable auto incrementing versions locally by setting the `autoIncrement` option on a build profile.
```

- Let's split this into two paragraphs.
- Let's simplify what we have to say about setting `appVersionSource` to `local`. The code block provides the JSON syntax.
